dandiapi/api/services/embargo/__init__.py

```diff
@@ -33,28 +39,66 @@ def _delete_asset_blob_tags(client: S3Client, blob: str):
 
 # NOTE: In testing this took ~2 minutes for 100,000 files
 def _remove_dandiset_asset_blob_embargo_tags(dandiset: Dandiset):
-    # First we need to generate a CSV manifest containing all asset blobs that need to be untaged
-    embargoed_asset_blobs = AssetBlob.objects.filter(
-        embargoed=True, assets__versions__dandiset=dandiset
-    ).values_list('blob', flat=True)
+    embargoed_asset_blobs = list(
+        AssetBlob.objects.filter(embargoed=True, assets__versions__dandiset=dandiset).values_list(
+            'blob', flat=True
+        )
+    )
 
     client = get_boto_client(config=Config(max_pool_connections=100))
     with ThreadPoolExecutor(max_workers=100) as e:
-        for blob in embargoed_asset_blobs:
+        futures = [
             e.submit(_delete_asset_blob_tags, client=client, blob=blob)
+            for blob in embargoed_asset_blobs
+        ]
```

Comment:
Should we considering batching these using `.iterator()` and `ichunked`? The Heroku worker dyno has 512MB of memory, so I wonder if this would use too much memory given enough assets

Reply:
Also, since the `ThreadPoolExecutor` has a max_workers of 100, it seems there would be no downside to batching things provided that the batch size is >= 100

Reply:
I think it's unlikely since we just return the `blob` value, but nevertheless it's better to err on the side of caution, to reduce memory consumption if possible.

Reply:
I had a similar thought about the possible number of embargoed assets in a given Dandiset.

Reply:
[566295a](https://github.com/dandi/dandi-archive/pull/1965/commits/566295a9c2461cc2bdecc1cbc4224c46c834651e). I ended up using `chunked` instead of `ichunked`, since the chunk is accessed more than once.